src/file/format/ron.rs

```diff
@@ -0,0 +1,74 @@
+use std::collections::HashMap;
+use std::error::Error;
+
+use ron;
+
+use crate::value::{Value, ValueKind};
+
+pub fn parse(
+    uri: Option<&String>,
+    text: &str,
+) -> Result<HashMap<String, Value>, Box<dyn Error + Send + Sync>> {
+    let value = from_ron_value(uri, &ron::from_str(text)?)?;
+    match value.kind {
+        ValueKind::Table(map) => Ok(map),
+
+        _ => Ok(HashMap::new()),
+    }
+}
+
+fn from_ron_value(
+    uri: Option<&String>,
+    value: &ron::Value,
+) -> Result<Value, Box<dyn Error + Send + Sync>> {
+    Ok(match *value {
+        ron::Value::Option(ref value) => match value {
+            Some(value) => {
+                let value = from_ron_value(uri, value.as_ref())?;
+                Value::new(uri, value.kind)
+            }
+            None => Value::new(uri, ValueKind::Nil),
+        },
+
+        ron::Value::Unit => Value::new(None, ValueKind::Nil),
+
+        ron::Value::Bool(value) => Value::new(uri, ValueKind::Boolean(value)),
+
+        ron::Value::Number(ref value) => {
+            if let Some(value) = value.as_i64() {
+                Value::new(uri, ValueKind::Integer(value))
+            } else if let Some(value) = value.as_f64() {
+                Value::new(uri, ValueKind::Float(value))
+            } else {
+                unreachable!();
+            }
+        }
+
+        ron::Value::Char(ref value) => Value::new(uri, ValueKind::String(value.to_string())),
+
+        ron::Value::String(ref value) => Value::new(uri, ValueKind::String(value.clone())),
+
+        ron::Value::Seq(ref values) => {
+            let mut array = Vec::new();
+
+            for value in values.iter() {
+                array.push(from_ron_value(uri, value)?);
+            }
+
+            Value::new(uri, ValueKind::Array(array))
+        }
+
+        ron::Value::Map(ref values) => {
+            let mut map = HashMap::new();
+
+            for (key, value) in values.iter() {
+                map.insert(
+                    key.clone().into_rust::<String>()?,
+                    from_ron_value(uri, value)?,
+                );
+            }
```

Comment:
Same as above applies here.

Reply:
Done, but the cloning unfortunately stayed because `ron::Map` does not implement `into_iter`, so I couldn't do much about that.

Reply:
ah, okay. So ignore that part of my latest comment please :laughing: